tests/fixtures/endpoints_handler.py

```diff
@@ -0,0 +1,205 @@
+            parameter_pattern,
+            repl=lambda name: f"(?P<{name.group().strip(strip)}>[^/]+)",
+            string=url_pattern,
+        )
+        return url
+
+    @classmethod
+    def _add_endpoint(cls, method: str, url: str, endpoint_function: Callable):
+        """Process url and store endpoint according to method type"""
+        url_pattern = cls._compile_regex_url(url)
+        matchable_endpoint = MatchableEndpoint(
+            url_pattern=url_pattern,
+            method=method,
+            endpoint_function=endpoint_function,
+        )
+        cls._methods[method].append(matchable_endpoint)
+        cls._methods[method].sort(
+            key=lambda endpoint: len(endpoint.url_pattern), reverse=True
+        )
+
+    @classmethod
+    def get(cls, url: str):
+        """Decorator function to add endpoint to Handler"""
+
+        def inner(endpoint_function: Callable):
+            cls._add_endpoint(
+                method="GET", url=url, endpoint_function=endpoint_function
+            )
+            return endpoint_function
+
+        return inner
+
+    @classmethod
+    def post(cls, url: str):
+        """Decorator function to add endpoint to Handler"""
+
+        def inner(endpoint_function: Callable):
+            cls._add_endpoint(
+                method="POST", url=url, endpoint_function=endpoint_function
+            )
+            return endpoint_function
+
+        return inner
+
+    @classmethod
+    def patch(cls, url: str):
+        """Decorator function to add endpoint to Handler"""
+
+        def inner(endpoint_function: Callable):
+            cls._add_endpoint(
+                method="PATCH", url=url, endpoint_function=endpoint_function
+            )
+            return endpoint_function
+
+        return inner
+
+    @staticmethod
+    def _convert_parameter_types(
+        endpoint_function: Callable,
+        string_parameters: dict[str, str],
+        request: httpx.Request,
+    ) -> dict[str, Any]:
+        """Get type info for function parameters. Since the values parsed from the URL
+        are still in string format, cast them to the types specified in the signature.
+        If the request is needed, include that in the returned parameters"""
+
+        # Get the parameter information from the endpoint function signature
+        signature_parameters = get_type_hints(endpoint_function)
+
+        # type-cast based on type-hinting info
+        typed_parameters: dict[str, Any] = {}
+        for parameter_name, value in string_parameters.items():
+            try:
+                parameter_type = signature_parameters[parameter_name]
+
+            # all parameters should be typed, raise exception otherwise
+            except KeyError:
+                raise TypeError(
+                    f"Parameter '{parameter_name}' in function "
+                    + f"'{endpoint_function.__name__}' is missing type information!"
+                )  # pylint: disable=broad-exception-raised
```

Comment:
```suggestion
                )
```
This is no longer needed, right?

Reply:
Nope, thanks: [ce0f140](https://github.com/ghga-de/ghga-connector/pull/68/commits/ce0f140127d1b8ada78cbcfc4fd23cb720965e6a)